### PR TITLE
LITE-27307: Add casting to formula transformation

### PR DESCRIPTION
**Pull request by gab832**

### CODE REVIEW

File: connect_transformations/formula/utils.py [hunk 1]
```diff
@@ -56,17 +66,8 @@ def validate_formula(data):  # noqa: CCR001
         if expression['to'] in output_columns:
             return error_response('Each `output column` must be unique.')
 
-        if expression['to'] in available_columns:
-            output_column = list(filter(
```
Comment (bdjilka) [inline connect_transformations/formula/utils.py:60]: This check must be kept

File: ui/src/pages/transformations/formula.js [hunk 2]
```diff
@@ -17,17 +18,49 @@ import {
...
-      <input type="text" placeholder="Output column" style="width: 70%;" ${output ? `value="${output}"` : ''} />
+      <div class="output-group">
+      <input id="output-${index}" type="text" placeholder="Output column" ${output ? `value="${output}"` : ''} />
+      ${typeSelect}
+      ${precisionSelect}
       <button id="delete-${index}" class="button delete-button">DELETE</button>
+      </div>
       <div class="input-group">
-          <label class="label" for="${columnId || `formula-${index}`}">Formula:</label>
-          <textarea id="${columnId || `formula-${index}`}" style="width: 100%;">${formula ? `${formula}` : ''}</textarea>
```
Comment (bdjilka) [inline ui/src/pages/transformations/formula.js:30]: Id is needed to pass to backend to identify whether formula is edited

**Review by bdjilka — CHANGES_REQUESTED** [review]

File: connect_transformations/formula/utils.py [hunk 3]
```diff
@@ -44,9 +44,19 @@ def validate_formula(data):  # noqa: CCR001
             or not expression['formula']
             or 'formula' not in expression
             or not isinstance(expression['formula'], str)
+            or 'type' not in expression
+            or expression['type'] not in [
```
[the remark this answers is not in the document]
Comment (gab832) [reply connect_transformations/formula/utils.py]: I agree, thanks!

File: ui/src/pages/transformations/formula.js (hunk 2 above)
Comment (gab832) [reply ui/src/pages/transformations/formula.js:30]: Yes sure, but I detected that if we change the name, the column is recreated in the backend side.

**Review by ffaraone — APPROVED** [review]

**Pull request merged**